3-async-await/solution.js

```diff
@@ -1,35 +1,73 @@
-/*
-INSTRUCTIONS
-
-1. using async/await API consume products and prices methods
-2. don't use .then(), .catch() or .finally() here
-3. both, products and prices methods expect a positive integer id
-4. use Promise.all() and Promise.allSettled() to consume both methods in parallel
-5. to generate the id do the following: invoke Date.now(), and take the last two digits, this will be your id
-6. log the results with console.log(), the format is up to you, but it must include id, product and price
-
-Example:
-{
- id:100,
- product:'paper',
- price:1
-}
+import products from "./products.js";
+import prices from "./prices.js";
+
+const generateID = () => new Date().getTime().toString().slice(-2);
+
+async function solution() {
+  const ID = generateID();
+
+  // Promises
+  const promises = [products(ID), prices(ID)];
+
+  // Promise.All
+  try {
+    const [productName, productPrice] = await Promise.all(promises);
+    console.log(`(Promise.All):`, {
+      id: ID,
+      product: productName,
+      price: productPrice,
+    });
+  } catch (error) {
+    console.log(`Error (Promise.All): ${error.message || "Unknown error"}`);
+  }
+
+  // Promise.allSettled
+  const resultsAllSettled = await Promise.allSettled(promises);
 
-7. both methods include some conditions to fail, at the end you should console.log() the errors, the format is up to you
-8. add any needed adjustment to solution() function
-9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
-*/
+  const allFulfilled = resultsAllSettled.every(
+    (result) => result.status === "fulfilled"
+  );
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+  if (allFulfilled) {
+    const productName = resultsAllSettled[0].value;
+    const productPrice = resultsAllSettled[1].value;
 
-    // You generate your id value here
+    console.log(`(Promise.AllSettled):`, {
+      id: ID,
+      product: productName,
+      price: productPrice,
+    });
+  }
 
-    // You use Promise.all() here
+  if (!allFulfilled) {
+    console.log(
+      `Error (Promise.allSettled): ${
+        resultsAllSettled[0].reason || "Unknown error"
+      }`
+    );
+  }
```

Comment:
We can take the `first error approach`, where we first have the error validation and then our code without errors. That improves the readability of the code.